sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AuthenticationProvider.java

```diff
@@ -0,0 +1,170 @@
+package com.microsoft.appcenter.analytics;
+
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.HandlerUtils;
+import com.microsoft.appcenter.utils.HashUtils;
+import com.microsoft.appcenter.utils.TicketCache;
+
+import java.util.Date;
+
+import static com.microsoft.appcenter.analytics.Analytics.LOG_TAG;
+
+public class AuthenticationProvider {
+
+    /**
+     * Ratio threshold of expiry time to refresh token a bit before it's expired.
```

Comment:
Current wording doesn't really make sense... I would go with "Ratio of refresh interval to token lifetime.  Used to refresh token a bit before it's expired."

Reply:
Updated